### PR TITLE
Blog App - processing data in models

### PR DESCRIPTION
**Pull request by tugsem**
## In this branch I: 
- Created model classes for all entities as shown in the [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png).
- Set up associations between models.
- Wrote methods for all the models.

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -1,4 +1,15 @@
 class Post < ApplicationRecord
-  has_many :comments, :likes
-  belongs_to :user, counter_cache: true
+  has_many :comments
+  has_many :likes
+  belongs_to :user, foreign_key: 'user_id', counter_cache: true, optional: true
+
+  after_save :update_post_counter
```
Comment (danmainah) [inline app/models/post.rb]: - [x] Kindy check on your method spelling its different from the update_counter method you created.

File: app/models/comment.rb [hunk 2]
```diff
@@ -1,4 +1,10 @@
 class Comment < ApplicationRecord
-  belongs_to :user
-  belongs_to :post, counter_cache: true
+  belongs_to :user, optional: true
+  belongs_to :post, counter_cache: true, optional: true
+
+  after_save :update_counter
+
+  def update_counter
```
Comment (danmainah) [inline app/models/comment.rb:7]: - [ ] [optional] It would be nice making this method private.

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @tugsem , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️ No linter errors
✔️ Gitflow was correctly used


# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by VICTOR-CHUKWU — APPROVED** [review]
Hi @tugsem , 
# STATUS: APPROVED! 🥇 💯 
done

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Davidosky007 — COMMENTED** [review]
Hi @tugsem 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
* No linter errors 👍 
* Well PR description 👍 
* Good looking readme file 👍 
* Proper use of Gitflow 👍 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**